### PR TITLE
Changed FKs and related_names in Circuits app

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #3068 
# What's Changed

1. Changed `type` FK to `circuit_type`, `termination_a` FK to `circuit_termination_a` and `termination_z` FK to `circuit_termination_z` on Circuit model.
2. Changed `circuit` FK and `_path` FK related_names to `circuit_terminations` on CircuitTermination model.
3. Changed related Forms, Filtersets, Serializers and Templates
4. Fixed failing unitests in `Circuits` and `DCIM` apps

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/3068.changed
Comment (glennmatthews) [inline changes/3068.changed:2]: Perhaps something like:

```suggestion
Renamed fields on Circuit model: `type` to `circuit_type`, `terminations` to `circuit_terminations`, `termination_a` to `circuit_termination_a`, and `termination_z` to `circuit_termination_z`.
Renamed reverse-relation `circuittermination` to `circuit_terminations` on the CablePath model.
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:43]: This should all be included in `Renamed Database Fields` above, IMO.

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:42]: IIRC the `_path` related-name was previously `circuittermination`?

File: nautobot/circuits/filters.py
Comment (glennmatthews) [inline nautobot/circuits/filters.py:157]: Can/should we go ahead and collapse these two to a `NaturalKeyOrPKMultipleChoiceFilter`?

File: nautobot/circuits/migrations/0010_rename_foregin_keys_and_related_names.py
Comment (glennmatthews) [inline nautobot/circuits/migrations/0010_rename_foregin_keys_and_related_names.py:1]: filename nit: `foregin` -> `foreign`.

File: nautobot/circuits/filters.py
Comment (HanlinMiao) [reply nautobot/circuits/filters.py:157]: Would this be out of scope since this [story](https://github.com/nautobot/nautobot/issues/2808) should be covering this change?

File: nautobot/circuits/models.py
Comment (gsnider2195) [inline nautobot/circuits/models.py:322]: Could we change this on the abstract base class `PathEndpoint` to the below reverse relation name? We need to document that as the standard for abstract base classes.

```py
related_name="%(app_label)s_%(class)s_related"
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (gsnider2195) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:26]:
```suggestion
|               | `termination_z`        | `circuit_termination_z`  |
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (gsnider2195) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:79]:
```suggestion
|               | `termination_z`  | `circuit_termination_z`  |  
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.0.md:61]:
```suggestion
Check out more Foregin Key related changes documented in this table [Renamed Foreign Key Fields](../installation/upgrading-from-nautobot-v1.md#renamed-foreign-key-fields)
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.0.md:68]:
```suggestion
Circuit.objects.filter(terminations__site__in=["ams01", "ams02", "atl03"])
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.0.md:74]:
```suggestion
Circuit.objects.filter(circuit_terminations__site__in=["ams01", "ams02", "atl03"])
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.0.md:77]:
```suggestion
Check out more `related-name` changes documented in this table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields)
```

**Comment by gsnider2195** [comment]
I think I may have found a line that needs to be updated as part of this change:

```
nautobot/core/constants.py:76:                "queryset": Circuit.objects.prefetch_related("type", "provider", "tenant", "terminations__site"),
```

File: nautobot/docs/configuration/optional-settings.md
Comment (gsnider2195) [inline nautobot/docs/configuration/optional-settings.md:1018]: Possible typo?

File: nautobot/docs/development/best-practices.md
Comment (gsnider2195) [inline nautobot/docs/development/best-practices.md:14]: Thanks! 👍

File: nautobot/circuits/filters.py
Comment (glennmatthews) [reply nautobot/circuits/filters.py]: That's fine, sure. Probably good to keep this more narrowly scoped.

File: nautobot/circuits/tests/test_filters.py
Comment (glennmatthews) [inline nautobot/circuits/tests/test_filters.py:220]: While you're touching this test case, would you mind changing it to use `assertQuerysetEqualAndNotEmpty`?

File: nautobot/circuits/views.py
Comment (glennmatthews) [inline nautobot/circuits/views.py:100]: Gonna be a lot of merge conflicts between this and #3112, will need to handle those carefully.

File: nautobot/circuits/models.py
Comment (glennmatthews) [inline nautobot/circuits/models.py:199]: This change will impact the CircuitFactory that I'm adding in #3115, just making a note here.

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:33]: Alphabetization:

```suggestion
| Circuit              | `termination_a`                         | `circuit_termination_a`                        |
|                      | `termination_z`                         | `circuit_termination_z`                        |
|                      | `terminations`                          | `circuit_terminations`                         |
|                      | `type`                                  | `circuit_type`                                 |
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:97]: Alphabetization:

```suggestion
| Circuit               | `termination_a`        | `circuit_termination_a`       |
|                       | `termination_z`        | `circuit_termination_z`       |
|                       | `type`                 | `circuit_type`                |
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:99]:
```suggestion
|                       | `type`                 | `cluster_type`                |
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:132]:
```suggestion
| Circuit               | `type`                    | `circuit_type`                   | `/circuits/circuits/?circuit_type=<uuid/slug>`                            |
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:161]:
```suggestion
| Circuit               | `circuit_type`       | Enhanced to support primary key UUIDs in addition to slugs | `/circuits/circuits/?circuit_type=<uuid/slug>`|
```

File: nautobot/docs/installation/upgrading-from-nautobot-v1.md
Comment (glennmatthews) [inline nautobot/docs/installation/upgrading-from-nautobot-v1.md:220]:
```suggestion
| Circuit            | `type_id`            | instead of `/circuits/circuits/?type_id=<uuid>`, use `circuit_type=<uuid>`            |
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:61]:
```suggestion
Check out more Foreign Key related changes documented in the table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields).
```

File: nautobot/docs/release-notes/version-2.0.md
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:77]:
```suggestion
Check out more `related-name` changes documented in the table [Renamed Database Fields](../installation/upgrading-from-nautobot-v1.md#renamed-database-fields).
```

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
Sorry we missed merging this earlier - looks like there are a bunch of merge conflicts now, can you take a look?

**Comment by HanlinMiao** [comment]
> Sorry we missed merging this earlier - looks like there are a bunch of merge conflicts now, can you take a look?

Yeah I will, I was gonna merge this after I merge the tenancy one, looks like that one also has a bunch of conflicts.

**Pull request merged**
